tests/flytekit/unit/core/test_dynamic.py

```diff
@@ -153,3 +153,15 @@ def wf() -> str:
         return dynamic_wf()
 
     assert wf() == "hello"
+
+
+def test_dynamic_return_dict():
```

Comment:
Can you add a few more tests? For example, at least one test where you return a tuple of dictionaries, also nested dictionaries.